src/containers/projectcontainer.js

```diff
@@ -29,46 +35,67 @@ class ProjectContainer extends Component {
     }
   }
 
-
-  onProjectUpdate = (data) => {
-    if (this.props.project.data === null) {
+  onSubmit = (e) => {
+    e.preventDefault();
+    const data = this.props.form.data;
+    // TODO: Use selected_project instead of props.params.id?
+    const id = this.props.params.id;
+    if (id === undefined) {
       this.props.createProject(data)
         .then(p => this.context.router.push(`/projects/${p.payload.id}`));
     } else {
-      this.props.updateProject(this.props.project.data.id, data)
-        .then(() => this.context.router.push(`/projects/${this.props.project.data.id}`));
+      this.props.updateProject(id, data);
     }
   }
 
+  onChange = (fieldName, value) => {
+    this.props.updateField(fieldName, value);
+  }
+
   render() {
-    return (
-      <Project
-        project={this.props.project}
-        customers={this.props.customers}
-        onProjectUpdate={this.onProjectUpdate}
-      />
-    );
+    const isNew = this.props.params.id === undefined;
+
+    if (this.props.customers.loading === true ||
+        this.props.form.data === undefined ||
```

Comment:
Can this happen? The initial state of `form` (in the reducer) isn't `undefined`.